configure.ac

```diff
@@ -45,9 +49,12 @@ AS_IF([test "$P4_PREFIX" = ""], [
   AC_SUBST([P4_PREFIX])
 ])
 
-diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/test.p4)
-AM_CONDITIONAL([BMV2_TEST_ENV],
-  [test "$P4_NAME" = "test" && test "$P4_PREFIX" = "test" && test "$diff_w_test" = "no"])
+diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/pdtest/pdtest.p4)
+
+if test "$diff_w_test" = "yes" ; then
+    diff_w_test=$($PYTHON $srcdir/tools/diff.py $P4_PATH $srcdir/bmv2/tests/openflow/oftest.p4)
+fi
+AM_CONDITIONAL([BMV2_TEST_ENV], [test "$P4_NAME" = "pdtest" || test "$P4_NAME" = "oftest" && test "$P4_PREFIX" = "pdtest" || test "$P4_PREFIX" = "oftest" && test "$diff_w_test" = "no"])
```

Comment:
I thought a little bit about it, and I think it maybe more readable to define 2 conditionals instead of one. Something like this:
```
diff_w_pdtest=
AM_CONDITIONAL([BMV2_PDTEST_ENV], [test "$P4_NAME" = "test" &&...])
diff_w_oftest=
AM_CONDITIONAL([BMV2_OFTEST_ENV], [test "$want_of" = yes && test "$P4_NAME" = "test" &&...])
```
This would make `tests/Makefile.am` pretty nice to:
```
if BMV2_PDTEST_ENV
SUBDIRS = pdtest
endif

if BMV2_ODTEST_ENV
SUBDIRS = oftest
endif

if !BMV2_PDTEST_ENV
if !BMV2_ODTEST_ENV
check-local: ... // the error message
endif
endif
```

What do you think?